src/osd/modules/input/input_evdev.cpp

```diff
@@ -0,0 +1,713 @@
+	{ BTN_BASE2,		ITEM_ID_INVALID,	"BASE2" },
+	{ BTN_BASE3,		ITEM_ID_INVALID,	"BASE3" },
+	{ BTN_BASE4,		ITEM_ID_INVALID,	"BASE4" },
+	{ BTN_BASE5,		ITEM_ID_INVALID,	"BASE5" },
+	{ BTN_BASE6,		ITEM_ID_INVALID,	"BASE6" },
+	{ BTN_DEAD,		ITEM_ID_INVALID,	"DEAD" },
+	{ BTN_SOUTH,		ITEM_ID_INVALID,	"A" },
+	{ BTN_EAST,		ITEM_ID_INVALID,	"B" },
+	{ BTN_C,		ITEM_ID_INVALID,	"C" },
+	{ BTN_NORTH,		ITEM_ID_INVALID,	"X" },
+	{ BTN_WEST,		ITEM_ID_INVALID,	"Y" },
+	{ BTN_Z,		ITEM_ID_INVALID,	"Z" },
+	{ BTN_TL,		ITEM_ID_INVALID,	"LEFT TRIGGER" },
+	{ BTN_TR,		ITEM_ID_INVALID,	"RIGHT TRIGGER" },
+	{ BTN_TL2,		ITEM_ID_INVALID,	"LEFT TRIGGER 2" },
+	{ BTN_TR2,		ITEM_ID_INVALID,	"RIGHT TRIGGER 2" },
+	{ BTN_SELECT,		ITEM_ID_INVALID,	"SELECT" },
+	{ BTN_START,		ITEM_ID_INVALID,	"START" },
+	{ BTN_MODE,		ITEM_ID_INVALID,	"MODE" },
+	{ BTN_THUMBL,		ITEM_ID_INVALID,	"LEFT THUMB" },
+	{ BTN_THUMBR,		ITEM_ID_INVALID,	"RIGHT THUMB" },
+	{ BTN_GEAR_DOWN,	ITEM_ID_INVALID,	"GEAR DOWN" },
+	{ BTN_GEAR_UP,		ITEM_ID_INVALID,	"GEAR UP" },
+	{ 0, ITEM_ID_INVALID, 0 },
+};
+
+idmap relaxis_map[] = {
+	{ REL_X,		ITEM_ID_XAXIS,		"X" },
+	{ REL_Y,		ITEM_ID_YAXIS,		"Y" },
+	{ REL_Z,		ITEM_ID_ZAXIS,		"Z" },
+	{ REL_RX,		ITEM_ID_RXAXIS,		"RX" },
+	{ REL_RY,		ITEM_ID_RYAXIS,		"RY" },
+	{ REL_RZ,		ITEM_ID_RZAXIS,		"RZ" },
+	{ REL_HWHEEL,		ITEM_ID_ADD_RELATIVE1,	"HWHEEL" },
+	{ REL_DIAL,		ITEM_ID_SLIDER2,	"DIAL" },
+	{ REL_WHEEL,		ITEM_ID_SLIDER1,	"WHEEL" },
+	{ REL_MISC,		ITEM_ID_ADD_RELATIVE2,	"EXTRA AXIS" },
+	{ REL_WHEEL_HI_RES,	ITEM_ID_SLIDER1,	"WHEEL" },
+	{ REL_HWHEEL_HI_RES,	ITEM_ID_ADD_RELATIVE1,	"HWHEEL" },
+	{ 0, ITEM_ID_INVALID, 0 },
+};
+
+idmap absaxis_map[] = {
+	{ ABS_X,		ITEM_ID_XAXIS,		"X" },
+	{ ABS_Y,		ITEM_ID_YAXIS,		"Y" },
+	{ ABS_Z,		ITEM_ID_ZAXIS,		"Z" },
+	{ ABS_RX,		ITEM_ID_RXAXIS,		"RX" },
+	{ ABS_RY,		ITEM_ID_RYAXIS,		"RY" },
+	{ ABS_RZ,		ITEM_ID_RZAXIS,		"RZ" },
+	{ ABS_THROTTLE,		ITEM_ID_SLIDER1,	"THROTTLE" },
+	{ ABS_RUDDER,		ITEM_ID_SLIDER2,	"RUDDER" },
+	{ ABS_WHEEL,		ITEM_ID_ADD_ABSOLUTE1,	"WHEEL" },
+	{ ABS_GAS,		ITEM_ID_ADD_ABSOLUTE2,	"GAS" },
+	{ ABS_BRAKE,		ITEM_ID_ADD_ABSOLUTE3,	"BRAKE" },
+	{ ABS_HAT0X,		ITEM_ID_ADD_ABSOLUTE4,	"HAT1 X" },
+	{ ABS_HAT0Y,		ITEM_ID_ADD_ABSOLUTE5,	"HAT1 Y" },
+	{ ABS_HAT1X,		ITEM_ID_ADD_ABSOLUTE6,	"HAT2 X" },
+	{ ABS_HAT1Y,		ITEM_ID_ADD_ABSOLUTE7,	"HAT2 Y" },
+	{ ABS_HAT2X,		ITEM_ID_ADD_ABSOLUTE8,	"HAT3 X" },
+	{ ABS_HAT2Y,		ITEM_ID_ADD_ABSOLUTE9,	"HAT3 Y" },
+	{ ABS_HAT3X,		ITEM_ID_ADD_ABSOLUTE10,	"HAT4 X" },
+	{ ABS_HAT3Y,		ITEM_ID_ADD_ABSOLUTE11,	"HAT4 Y" },
+	{ ABS_PRESSURE,		ITEM_ID_ADD_ABSOLUTE12,	"PRESSURE" },
+	{ ABS_DISTANCE,		ITEM_ID_ADD_ABSOLUTE13,	"DISTANCE" },
+	{ ABS_TILT_X,		ITEM_ID_ADD_ABSOLUTE14,	"TILT X" },
+	{ ABS_TILT_Y,		ITEM_ID_ADD_ABSOLUTE15,	"TILT Y" },
+	{ ABS_TOOL_WIDTH,	ITEM_ID_ADD_ABSOLUTE16,	"WIDTH" },
+	{ 0, ITEM_ID_INVALID, 0 },
+};
+
+enum device_type {
+	EVDEV_NONE,
+	EVDEV_KEYBOARD,
+	EVDEV_MOUSE,
+	EVDEV_JOYSTICK,
+	EVDEV_GENERIC,
+};
+
+
+#define OCTETS_FOR(n) ((n+7)>>3)
```

Comment:
Please don’t use macros unnecessarily – you can use `constexpr` functions for things like this.

Reply:
Gotcha.